WordPress/openverse
# Expose provider in the API tags response

## Fixes



Fixes #4273 by @AetherUnbound

## Description



This PR exposes the `provider` field within the tags attribute that is present in the database under the key `unstable__provider`. The core of this change is in `media_serializers.py`, most of the other changes in this PR revolve around making sure the field shows up in tests and documentation.

One thing that was unexpected was having to set `allow_null=True` on the serializer. It seems like there are some cases where we're converting results directly from Elasticsearch into serialized results rather than retrieving the full data from the database. We could alter the ingestion server to ingest this new field, but I don't think it's necessary (and I think we can leave this as nullable here, given the explanation I've provided).


## Testing Instructions


Tests should pass! You can also do the following:

1. `just down -v`
1. `just api/init`
1. Visit http://localhost:50280/v1/images/ and see that each of the tags arrays also sends an `unstable__provider` key
1. Visit http://localhost:50280/v1/images/?q=crim and observe a result that has tags from multiple providers


## Checklist


      PRs) or the media properties generator (`just catalog/generate-docs media-props`
      for the catalog or `just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
